### PR TITLE
Added nodeSelector support

### LINKED ISSUE #762: Add support for `nodeSelector`s to the CRDB statefulset
Am I missing it or has this simply not been implemented yet? Is this a desired feature? TBH, I'm not sure how you're supposed to control how these pods are scheduled without nodeSelectors (or affinities/tolerations).

### PR DESCRIPTION
**Pull request by cin**
Added support for nodeSelectors to the CRD.

Resolves #762

### CODE REVIEW

**Review by chrislovecnm — COMMENTED** [review]
Thanks for the PR.  Some comments for you.

File: pkg/resource/statefulset.go [hunk 1]
```diff
@@ -211,6 +211,7 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 			Containers:                    b.MakeContainers(),
 			AutomountServiceAccountToken:  ptr.Bool(false),
 			ServiceAccountName:            "cockroach-database-sa",
+			NodeSelector:                  b.Spec().NodeSelector,
```
[the remark this answers is not in the document]
Comment (cin) [reply pkg/resource/statefulset.go:214]: Just to be clear, do you mean something like:
```go
	if b.Spec().NodeSelector != nil && len(b.Spec().NodeSelector) > 0 {
		pod.Spec.NodeSelector = b.Spec().NodeSelector
	}
```

Or do you mean put this behind a feature gate?

[the remark this answers is not in the document]
Comment (chrislovecnm) [reply pkg/resource/statefulset.go:214]: The if statement is perfect

File: examples/3-nodes-insecure-with-nodeselector.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 60Gi
+        volumeMode: Filesystem
+  image:
+    name: cockroachdb/cockroach:v20.2.5
+  nodes: 3
```
[the remark this answers is not in the document]
Comment (cin) [reply examples/3-nodes-insecure-with-nodeselector.yaml]: Good points. I didn't know about the example templates. I did as you suggested and regenerated the templates. LMK if there's anything else to do in this regard.

**Review by chrislovecnm — APPROVED** [review]
LGTM

File: config/manager/patches/image.yaml [hunk 3]
```diff
@@ -24,4 +24,4 @@ spec:
     spec:
       containers:
         - name: cockroach-operator
-          image: cockroachdb/cockroach-operator:v2.2.1
+          image: cockroachdb/cockroach-operator:2.2.1
```
Comment (pseudomuto) [inline config/manager/patches/image.yaml:27]: This shouldn't have changed. Did this happen when you ran make/gen-templates?

**Review by pseudomuto — CHANGES_REQUESTED** [review]

**Review by pseudomuto — APPROVED** [review]

**Comment by cin** [comment]
I rebased and squashed all the intermittent commits. Should be good to go now.

**Pull request merged**